packages/cli/src/commands/template/init-v2.ts

```diff
@@ -0,0 +1,332 @@
+      cdPrefix = `cd ${templateDirName} && `
+    } catch (error) {
+      // Handle the case where package.json does not exist
+      const createRoot = path.join(root, templateDirName)
+      pkgJson = await PackageJson.create(createRoot)
+    }
+
+    switch (files.language) {
+      case Language.TypeScript:
+        pkgJson.update({
+          scripts: {
+            ...pkgJson.content.scripts,
+            'e2b:build:dev': `${cdPrefix}npx tsx ${files.buildDevFile}`,
+            'e2b:build:prod': `${cdPrefix}npx tsx ${files.buildProdFile}`,
+          },
+        })
+        break
+      default:
+        throw new Error('Unsupported language for package.json scripts')
+    }
+
+    // Save the changes
+    await pkgJson.save()
+
+    console.log('\n📝 Added build scripts to package.json:')
+    console.log(
+      `   ${asPrimary('npm run e2b:build:dev')} - Build development template`
+    )
+    console.log(
+      `   ${asPrimary('npm run e2b:build:prod')} - Build production template`
+    )
+  } catch (err) {
+    console.warn(
+      '\n⚠️  Could not add scripts to package.json:',
+      err instanceof Error ? err.message : err
+    )
+  }
+}
+
+/**
+ * Check template name format
+ */
+function templateNameRegex(name: string): boolean {
+  return /^[a-z0-9][a-z0-9_-]*[a-z0-9]$|^[a-z0-9]$/.test(name)
+}
+
+function validateTemplateName(name: string) {
+  if (!name || name.trim().length === 0) {
+    throw new Error('Template name cannot be empty')
+  }
+  if (!templateNameRegex(name.trim())) {
+    throw new Error(
+      'Template name must contain only lowercase letters, numbers, hyphens, and underscores, and cannot start or end with a hyphen or underscore'
+    )
+  }
+}
+
+export const initV2Command = new commander.Command('init-v2')
+  .description('initialize a new sandbox template using the SDK')
+  .addOption(pathOption)
+  .option('-n, --name <name>', 'template name (alias)', (value) => {
+    try {
+      validateTemplateName(value)
+    } catch (err) {
+      throw new commander.InvalidArgumentError(
+        err instanceof Error ? err.message : String(err)
+      )
+    }
+
+    return value
+  })
+  .option(
+    '-l, --language <language>',
+    `target language: ${Object.values(Language).join(', ')}`,
+    (value) => {
+      if (!Object.values(Language).includes(value as Language)) {
+        throw new commander.InvalidArgumentError(
+          `Invalid language. Must be one of: ${Object.values(Language).join(
+            ', '
+          )}`
+        )
+      }
+      return value as Language
+    }
+  )
+  .alias('it-v2')
+  .alias('create')
+  .action(
+    async (opts: { path?: string; name?: string; language?: Language }) => {
+      try {
+        process.stdout.write('\n')
+
+        const root = getRoot(opts.path)
+
+        console.log('🚀 Initializing Sandbox Template...\n')
+
+        // Step 1: Get template name (from CLI or prompt)
+        let templateName: string = opts.name ?? DEFAULT_TEMPLATE_NAME
```

Comment:
I don't think you need `?? DEFAULT_TEMPLATE_NAME` because it will be set in the input? and if you clear the pre-defined name it should throw a validation error anyways